app/controllers/frontend/feeds_controller.rb

```diff
@@ -1,75 +1,132 @@
 module Frontend
   class FeedsController < FrontendController
     before_action :set_conference, only: %i(podcast_folder)
-    EXPIRE_FEEDS = 15
+    FEEDS_EXPIRY_DURATION = 15.minutes
 
-    # podcast_recent
     def podcast
-      time = round_time(Time.now.ago(2.years))
-      xml = Rails.cache.fetch([:podcast, time.to_i], expires_in: EXPIRE_FEEDS.minutes) do
-        Feeds::PodcastGenerator.create_preferred(
-          view_context: view_context,
-          title: 'recent events feed', summary: 'This feed contains events from the last two years',
-          logo: logo_image_url,
-          events: downloaded_events.newer(time))
+      events_max_age = round_to_quarter_hour(Time.now.ago(2.years))
+
+      xml = Rails.cache.fetch([:podcast, params[:quality], events_max_age.to_i], expires_in: FEEDS_EXPIRY_DURATION) do
+        feed = Feeds::PodcastGenerator.new(
+          view_context,
+          title: 'recent events feed',
```

Comment:
The Feed-Title should reflect the Quality, so that Feeds with different Content have a distinguishable name.